File: samples/client/petstore/java/src/main/java/org/openapijsonschematools/client/schemas/validation/PatternPropertiesValidator.java
package org.openapijsonschematools.client.schemas.validation;

import org.checkerframework.checker.nullness.qual.Nullable;

import java.util.Map;

public class PatternPropertiesValidator implements KeywordValidator {
    @Override
    public @Nullable PathToSchemasMap validate(
        ValidationData data
    ) {
        var patternProperties = data.schema().patternProperties;
        if (patternProperties == null) {
            return null;
        }
        if (!(data.arg() instanceof Map<?, ?>)) {
            return null;
        }
        return data.patternPropertiesPathToSchemas();
    }
}

File: samples/client/petstore/kotlin/src/main/kotlin/org/openapijsonschematools/client/schemas/validation/PatternPropertiesValidator.kt
package org.openapijsonschematools.client.schemas.validation

class PatternPropertiesValidator : KeywordValidator {
    override fun validate(
        data: ValidationData
    ): PathToSchemasMap?
    {
        if (data.schema.patternProperties == null) {
            return null
        }
        return if (data.arg !is Map<*, *>) {
            null
        } else data.patternPropertiesPathToSchemas
    }
}
